src/workerd/server/container-client.c++

```diff
@@ -106,13 +106,21 @@ ContainerClient::ContainerClient(capnp::ByteStreamFactory& byteStreamFactory,
     kj::Network& network,
     kj::String dockerPath,
     kj::String containerName,
-    kj::String imageName)
+    kj::String imageName,
+    kj::TaskSet& waitUntilTasks)
     : byteStreamFactory(byteStreamFactory),
       timer(timer),
       network(network),
       dockerPath(kj::mv(dockerPath)),
       containerName(kj::encodeUriComponent(kj::mv(containerName))),
-      imageName(kj::mv(imageName)) {}
+      imageName(kj::mv(imageName)),
+      waitUntilTasks(waitUntilTasks) {}
+
+ContainerClient::~ContainerClient() noexcept(false) {
+  waitUntilTasks.add(dockerApiRequest(network, kj::str(dockerPath), kj::HttpMethod::DELETE,
```

Comment:
It looks like this `dockerApiRequest()` will keep accessing members of `ContainerClient` after it has been destroyed. I think it needs to be refactored into a static function that can take ownership of the things it needs to complete itself.

Reply:
Right now, it only accesses kj::network. And we can't transfer ownership since, worker service is the owner, and will live longer than ContainerClient. Any suggestions?

Reply:
It was refactored to a static function though.